app/models/hint.rb

```diff
@@ -1,5 +1,52 @@
+require 'stringex/core_ext'
+
 class Hint < ApplicationRecord
   def self.match(searchterm)
-    Hint.find_by(fingerprint: searchterm)
+    searchprint = fingerprint(searchterm)
+    Hint.find_by(fingerprint: searchprint)
   end
-end
+
+  # This implements the OpenRefine fingerprinting algorithm. See
+  # https://github.com/OpenRefine/OpenRefine/wiki/Clustering-In-Depth#fingerprint
+  def self.fingerprint(searchterm)
+    # Preprocess.
+    searchterm.strip!
+    searchterm.downcase!
+
+    depunctuate!(searchterm)
```

Comment:
thanks for moving this to it's own method. much easier to follow this method with the 🙈 moved to `depunctuate!` :)